src/gui/src/dbDescriptors.cpp

```diff
@@ -1639,4 +1648,139 @@ bool DbTechLayerDescriptor::getAllObjects(SelectionSet& objects) const
   return true;
 }
 
+//////////////////////////////////////////////////
+
+static const char*
+getAccessTypeName(odb::dbAccessPoint::AccessType type)
+{
+  switch(type) {
+  case odb::dbAccessPoint::OnGrid:
+    return "OnGrid";
+  case odb::dbAccessPoint::HalfGrid:
+    return "HalfGrid";
+  case odb::dbAccessPoint::Center:
+    return "Center";
+  case odb::dbAccessPoint::EncOpt:
+    return "EncOpt";
+  case odb::dbAccessPoint::NearbyGrid:
+    return "NearbyGrid";
+  }
+  return "Unknown";
+}
+
+DbItermAccessPointDescriptor::DbItermAccessPointDescriptor(odb::dbDatabase* db) :
+    db_(db)
+{
+}
+
+std::string DbItermAccessPointDescriptor::getName(std::any object) const
+{
+  auto iterm_ap = std::any_cast<DbItermAccessPoint>(object);
+  auto ap = iterm_ap.ap;
+  std::string name(getAccessTypeName(ap->getLowType()));
+  name += std::string("/") + getAccessTypeName(ap->getHighType());
+  return name;
+}
+
+std::string DbItermAccessPointDescriptor::getTypeName() const
+{
+  return "Access Point";
+}
+
+bool DbItermAccessPointDescriptor::getBBox(std::any object, odb::Rect& bbox) const
+{
+  auto iterm_ap = std::any_cast<DbItermAccessPoint>(object);
+  odb::Point pt = iterm_ap.ap->getPoint();
+  odb::dbTransform xform;
+  int x, y;
+  iterm_ap.iterm->getInst()->getLocation(x, y);
+  xform.setOffset({x, y});
+  xform.apply(pt);
+  bbox = {pt, pt};
+  return true;
+}
+
+void DbItermAccessPointDescriptor::highlight(std::any object,
+                                        Painter& painter,
+                                        void* additional_data) const
+{
+  auto iterm_ap = std::any_cast<DbItermAccessPoint>(object);
+  odb::Point pt = iterm_ap.ap->getPoint();
+  odb::dbTransform xform;
+  int x, y;
+  iterm_ap.iterm->getInst()->getLocation(x, y);
+  xform.setOffset({x, y});
+  xform.apply(pt);
+  const int shape_size = 100;
+  painter.drawLine({pt.x() - shape_size / 2, pt.y() - shape_size / 2},
+                   {pt.x() + shape_size / 2, pt.y() + shape_size / 2});
+  painter.drawLine({pt.x() - shape_size / 2, pt.y() + shape_size / 2},
+                   {pt.x() + shape_size / 2, pt.y() - shape_size / 2});
```

Comment:
Since there are several places now in the GUI that needs to draw crosses, maybe it makes sense to add a function to Painter to support that in a central location. (DRC, the layout widget for access points are the two I can think of right now).

Reply:
Agreed